src/utils/solidity/transactionFactory.ts

```diff
@@ -15,9 +15,12 @@ import {
   Options,
 } from '../solidity';
 import { Address } from '../types';
+import { ensure } from '../guards';
 
-type TransactionArg = number | number[] | string | string[];
-type TransactionArgs = TransactionArg[];
+// TODO: Fix the types here once the transaction factory decorators are
```

Comment:
This can lead to weird problems: If the EVM receives other stuff than strings/numbers, it fails with misleading error messages.